### PR TITLE
refactor: use poetry

### PR DESCRIPTION
**Pull request by vanstinator**
Refactoring packaging and dependency management to use Poetry. I've been using this on another project lately and it's really nice keeping both install _and_ dev deps contained and easy to install. I need to write docs yet.

In addition to this I scaffolded a very basic unit test so that we can start building up from here as we add more features and generations that we don't support today.

### CODE REVIEW

**Comment by vanstinator** [comment]
> Interesting choices

Please refrain from jumping in with vague comments. If you have constructive feedback it's more than welcome 🙂

**Comment by toddrob99** [comment]
> > Interesting choices
> 
> Please refrain from jumping in with vague comments. If you have constructive feedback it's more than welcome 🙂

It wasn't really a vague comment when taken in the context of the [single line](https://github.com/toyotha/toyota-na/commit/55c7ebee64b50cf9deb9259a86320d7e9afbc2f1#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R13) the comment was related to... On the pull request itself I don't see my comment linked to that line, so I guess your response is fair.

> keywords = ["bluetooth", "melnor", "home automation", "irrigation", "gardening"]

Those keywords in pyproject.toml don't appear relevant for this project.

**Comment by vanstinator** [comment]
I didn't see the line reference in the review window nor the commit hash linked in the post body, which is weird. Github is drunk I guess.

Glad you caught the error. I just happened to see it while doing a self review later 😅. That's what I get for copying and pasting too fast.

**Comment by toddrob99** [comment]
No worries, and if I had known GitHub would display it that way, I definitely would have included more detail. I love this project (using with Home Assistant); keep up the good work!

File: tests/test_vehicle.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from toyota_na.vehicle.base_vehicle import ApiVehicleGeneration
+from toyota_na.vehicle.vehicle import get_vehicles
+
+
+@pytest.fixture
+async def client():
+
+    vehicle_list = open("./samples/user_vehicle_list.json", "r", encoding="utf-8")
+    vehicle_status = open("./samples/vehicle_status.json", "r", encoding="utf-8")
+    telemetry = open("./samples/vehicle_health_status.json", "r", encoding="utf-8")
```
Comment (jlmaners) [inline tests/test_vehicle.py:16]: I _think_ this is the wrong file to test against. I believe you need to pull down valid out put from the `/v2/telemetry` endpoint as this is different from the vehicle_health_status one. They return similar but slightly different data from what I can see for my vehicle generation (17CY) so I'd just encourage a double check there.

**Review by jlmaners — APPROVED** [review]
Other than double checking the sample file is correct, this looks like reasonable change to merge in and appears to make life a little easier on the build/release side.

File: tests/test_vehicle.py (hunk 1 above)
Comment (vanstinator) [reply tests/test_vehicle.py:16]: Sure thing 👍🏻

Comment (vanstinator) [reply tests/test_vehicle.py]: Good catch. Grabbed a sample for `telemetry`.

**Comment by vanstinator** [comment]
I've enabled `branch protections` for `master` as well to enforce passing tests. Admins can override easily on a case-by-case basis so for the two of us it's more of a helpful tool than anything.

**Pull request merged**